### DISCUSSION TITLE
Significant difference between Infracost and Azure calculator

### DISCUSSION THREAD

**Discussion by mloskot** [infracost/infracost#2336] [General]
I have a Terraform plan provisioning a new node pool for my AKS:

- VM size: Standard_E2as_v5 
- OS: Windows (set via `os_type` of [azurerm_kubernetes_cluster_node_pool](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster_node_pool))
- VM/Node count: 1
- Premium disk P10
- Pay as go
- Region: UK South

Here is the Infracost estimate calculated based on my Terraform plan:

```
+ module.aks_windows_node_pool.azurerm_kubernetes_cluster_node_pool.node_pools[3]
  +£84.93
    + Instance usage (pay as you go, Standard_E2as_v5)
      +£68.55
    + os_disk
        + Storage (P10, LRS)
          +£16.38
```

UPDATE: Suspecting `location` property of resources in my Terraform code may be not enough, I've added `INFRACOST_AZURE_OVERRIDE_REGION=uksouth` environment variable and Infracost gives a greater estimate - still lower than Azure calculator's:

```
+ module.aks_windows_node_pool.azurerm_kubernetes_cluster_node_pool.node_pools[4]
  +£101
    + Instance usage (pay as you go, Standard_E2as_v5)
      +£80.69
    + os_disk 
        + Storage (P10, LRS)
          +£19.82
```

If I go to https://azure.microsoft.com/en-gb/pricing/calculator/ and request estimate for the same components in the same region, then the total for VM + Disk is roughly twice higher, that the `£84.93` above:

![image](https://user-images.githubusercontent.com/80741/223424206-7a543cea-9064-410e-8da5-73c7b90f9800.png)

What is the reason of such a difference or where am I making the mistake?

**Comment by alikhajeh1** [comment]
@mloskot I'm sorry I missed this, I answered your [other](https://github.com/infracost/infracost/discussions/2311) discussion topic but somehow I missed this one. Is https://github.com/infracost/infracost/issues/2371 tracking this issue? If not, can you possibly create an issue with the TF code snippet you used to generate that Infracost output? That helps us investigate it quickly as we can reproduce the issue vs guessing what params might have resulted in that output 😅

**Reply by mloskot** [reply]
@alikhajeh1 The issue #2371 is follow-up tracking my other thread in https://github.com/infracost/infracost/discussions/2368

However, this thread here may be related to that problem too indeed. So, I'd rather wait until #2371 is fixed, then I test it.
If the estimate discrepancy in comparison to the calculator is still observed, then I will open new issue. Is that OK?

**Reply by alikhajeh1** [reply]
Thanks - sounds good!

**Reply by mloskot** [reply]
The new release is out, but it doesn't include a fix for #2371, so I'll keep this thread open.

**Reply by mloskot** [reply]
For the archive records, fix is available and it works, see https://github.com/infracost/infracost/issues/2371#issuecomment-1629491310
Thanks

**Reply by aliscott** [reply]
Thanks for confirming @mloskot 🙏
